chain-impl-mockchain/src/smartcontract/mod.rs

```diff
@@ -0,0 +1,239 @@
+//! EVM Smart Contract transactions
+
+use chain_core::mempack::{ReadError, Readable};
+#[cfg(feature = "evm")]
```

Comment:
This is general and not only for this particular import but I wonder if we should instead feature-gate e.g. the whole `smartcontract` module instead of the specific logic inside each function